doc/normalization.rst

```diff
@@ -0,0 +1,83 @@
+Gardener Component Descriptor Normalization
+===========================================
+
+Every resource can contain an optional digest attribut that specifies the digest of the referenced resource.
+The digest is a object that *MUST* contain the `algorithm` and the hash as `value`.
+
+.. code-block::
+  meta:
+    schemaVersion: 'v2'
+
+  component:
+    name: 'example.com/my-component'
+    version: 'v1.1.1'
+
+    resources:
+    - name: my-artifact
+      version: v0.0.0
+      digest:
+        algorithm: 'sha256'
+        value: 'abcdef'
+
+
+Normalization
+--------------
+
+The normalized component descriptor has the following properties:
+- Attributes *MUST* be transformed into list of single attributes (key-value pairs) and ordered alphabetically
+- The component descriptor *MUST* be JSON encoded
+- Resources contain their identity fields (name, version and extra identity) as well as their content digest.
+- Sources are included only their identity fields
+  - Sources without a dedicated artifact can be added to the normalized version by using the `extraIdentity` field.
+
+.. code-block::
+  - meta:
+    - schemaVersion: 'v2'
+
+  - component:
+      - name: 'example.com/my-component'
+      - version: 'v1.1.1'
+
```

Comment:
We need component references as well as part of the normalized form, including the hash (not signed) of it.
Otherwise, we would not sign referenced components, wdyt?

Reply:
yes correct.
We need the list of component references with their identity and hash. 
If the hash of the referenced component is not already calculated we have to recursively calculate it